misc-utils/kill.c

```diff
@@ -286,6 +307,25 @@ static char **parse_arguments(int argc, char **argv, struct kill_control *ctl)
 			ctl->use_sigval = 1;
 			continue;
 		}
+#endif
+#ifdef UL_HAVE_PIDFD
+		if (!strcmp(arg, "--timeout")) {
+			struct timeouts *next;
+
+			ctl->timeout = 1;
+			if (argc < 2)
+				errx(EXIT_FAILURE, _("option '%s' requires an argument"), arg);
+			argc--, argv++;
+			arg = *argv;
+			next = xcalloc(1, sizeof(*next));
+			next->period = strtos32_or_err(arg, _("argument error"));
+			argc--, argv++;
```

Comment:
You have to initialize the list too:
                          INIT_LIST_HEAD(&next->follow_ups);

and the same is necessary before you start to use ctl->follow_ups.